### ISSUE TITLE
Let's learn how to test!

### ISSUE THREAD

**Issue by frattezi** [fmagesty/primeiro-ciclo-de-mentoria#18]
Please read [this documentation](https://reactjs.org/docs/testing.html). Let's learn about the testing frameworks we can use, and maybe create a simple test!

Comment in this issues your perspectives and learnings about tests:

- What are unit testings, why we should do them?
- What is Jest and React Testing?
- How should you test your react applications

And anything else you find interesting!

**Comment by fmagesty** [comment]
working on it

**Comment by fmagesty** [comment]
the jest is already working, I'll be posting here the 'how and why' of doing tests later

**Comment by fmagesty** [comment]
So, we basically need to test to ensure what is called 'quality code'. Testing also makes it easier to debug code later if needed, since it removes potential bugs after we test properly. These are the main reasons of testing and why it is so important. 

Jest is a library for testing and it is also recommended by the React/Facebook team. There is also a React testing library but it doesn't have the mock as the same way that Jest does, so I would guess because of this and other reasons Jest is a better library for testing in our case. 

For testing I followed the Jest docs where we create the files sum.js and sum.test.js and then run `npm run test` to make the test.
I also just found a very good link explaining lots of test case options using Jest: https://jestjs.io/docs/tutorial-react
